provider/oci/environ_test.go

```diff
@@ -0,0 +1,1022 @@
+
+	ids, err := e.env.ControllerInstances(nil, e.controllerUUID)
+	c.Assert(err, gc.IsNil)
+	c.Check(len(ids), gc.Equals, 1)
+}
+
+func (e *environSuite) setupListInstancesExpectations(instanceId string, state ociCore.InstanceLifecycleStateEnum, times int) {
+	listInstancesRequest, listInstancesResponse := makeListInstancesRequestResponse(
+		[]ociCore.Instance{
+			{
+				AvailabilityDomain: makeStringPointer("fakeZone1"),
+				CompartmentId:      &e.testCompartment,
+				Id:                 &instanceId,
+				LifecycleState:     state,
+				Region:             makeStringPointer("us-phoenix-1"),
+				Shape:              makeStringPointer("VM.Standard1.1"),
+				DisplayName:        makeStringPointer("fakeName"),
+				FreeformTags:       e.tags,
+			},
+		},
+	)
+	expect := e.compute.EXPECT().ListInstances(
+		context.Background(), listInstancesRequest).Return(
+		listInstancesResponse, nil)
+	if times == 0 {
+		expect.AnyTimes()
+	} else {
+		expect.Times(times)
+	}
+}
+
+func (e *environSuite) setupStopInstanceExpectations(instanceId string) {
+	listInstancesRequest, listInstancesResponse := makeListInstancesRequestResponse([]ociCore.Instance{
+		{
+			AvailabilityDomain: makeStringPointer("fakeZone1"),
+			CompartmentId:      &e.testCompartment,
+			Id:                 &instanceId,
+			LifecycleState:     ociCore.InstanceLifecycleStateRunning,
+			Region:             makeStringPointer("us-phoenix-1"),
+			Shape:              makeStringPointer("VM.Standard1.1"),
+			DisplayName:        makeStringPointer("fakeName"),
+			FreeformTags:       e.tags,
+		},
+	})
+	requestMachine1, responseMachine1 := makeGetInstanceRequestResponse(ociCore.Instance{
+		CompartmentId:      listInstancesResponse.Items[0].CompartmentId,
+		AvailabilityDomain: listInstancesResponse.Items[0].AvailabilityDomain,
+		Id:                 listInstancesResponse.Items[0].Id,
+		Region:             listInstancesResponse.Items[0].Region,
+		Shape:              listInstancesResponse.Items[0].Shape,
+		DisplayName:        listInstancesResponse.Items[0].DisplayName,
+		FreeformTags:       listInstancesResponse.Items[0].FreeformTags,
+		LifecycleState:     ociCore.InstanceLifecycleStateRunning,
+	})
+
+	e.listInstancesResponse.RawResponse = &http.Response{
+		StatusCode: 200,
+	}
+	responseMachine1.RawResponse = &http.Response{
+		StatusCode: 200,
+	}
+
+	terminateRequestMachine1 := ociCore.TerminateInstanceRequest{
+		InstanceId: listInstancesResponse.Items[0].Id,
+	}
+
+	terminateResponse := ociCore.TerminateInstanceResponse{
+		RawResponse: &http.Response{
+			StatusCode: 201,
+		},
+	}
+
+	responseMachine1Terminating := responseMachine1
+	responseMachine1Terminating.Instance.LifecycleState = ociCore.InstanceLifecycleStateTerminating
+	responseMachine1Terminated := responseMachine1Terminating
+	responseMachine1Terminated.Instance.LifecycleState = ociCore.InstanceLifecycleStateTerminated
+
+	gomock.InOrder(
+		e.compute.EXPECT().ListInstances(
```

Comment:
If you are doing a lot of these, I do tend to prefer:
exp := e.compute.Expect()

gomock.InOrder(
  exp.ListInstances(
 ...
  exp.GetInstance(
...
etc.

Especially because EXPECT is in all caps, but *isn't* the thing that is interesting.

Reply:
Nice! will change.